### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by JethroWarnett**
Reworked alot of the code to make it cleaner, declarative and smaller. Removed unecessary code. Fixed some bugs in undoRedo and importExportservice. Changed the css of the web app. Removed ngStyle from HTML.

### CODE REVIEW

File: src/app/components/fairChain/fairChain.component.ts [hunk 1]
```diff
@@ -30,43 +31,64 @@ enum ChangingEdge {
...
 
   private changesNode: ChangingNode = ChangingNode.None;
   private changesEdge: ChangingEdge = ChangingEdge.None;
-  public currentTool: activeTool = activeTool.Idle;
+  private currentTool: activeTool = activeTool.Idle;
 
 
   // Create an array with nodes
-  private nodes: vis.Node[] = [];
+  //private nodes: Node[] = [];
```
Comment (GattoPianista) [inline src/app/components/fairChain/fairChain.component.ts:90]: Should be deleted

File: src/app/components/fairChain/fairChain.component.ts [hunk 2]
```diff
@@ -320,23 +259,11 @@ export class FairChainComponent implements OnInit {
    * Downloads the file as Graph.json with the method in importExport.service.
    */
   public exportGraph(){
-    this.updateNodePositions();
-    var text = "{\"nodes\":" + JSON.stringify(this.nodes) +",\"edges\":" + JSON.stringify(this.edges)+"}";
+    var text = this.importExportService.convertNetworkToJSON(this.nodes, this.edges);
     var filename = "Graph.json";
     this.importExportService.download(filename, text);
```
Comment (GattoPianista) [inline src/app/components/fairChain/fairChain.component.ts:284]: Maybe we could just replace a lot of this by making this.importExportService.download(this.nodes, this.edges), then it covertes it to JSON inside and also the file name  can be written inside ExportExportServie. But it is a minor thing

File: src/app/importExport.service.ts [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
     providedIn: 'root'
   })
 export class ImportExportService{
 
-  private nodes: vis.Node[] = [];
-  private edges: vis.Edge[] = [];
-  private data: vis.Data = {
+  private nodes: Node[] = [];
+  private edges: Edge[] = [];
+  private data: Data = {
```
Comment (GattoPianista) [inline src/app/importExport.service.ts]: Are we sure we can not use dataset here?

**Review by GattoPianista — APPROVED** [review]
Good work, we should fix some minor things, but you did a lot!

**Review by mdeliso97 — APPROVED** [review]
As Blendi alredy did a code review and didn't have much to add to his review (gg), I tested out the functionalities.
The versions seems to be buggy, while testing I encountered about 5 bugs:
- while changing node label, editedgemode is activated also, this might be caused by the changed condition in the onClick if statement (I solved it there previously)
- Import/export breaks the change label and color functionalities
- Edit edge label doesn't work anymore
- By editing the color of nodes, also in this case editEdgeMode is activated
- Redo/Undo also braks change color/label functionalities such like Import/Export
Maybe is better to do multiple branch bug fix for each problem before merging in dev

**Review by mdeliso97 — CHANGES_REQUESTED** [review]
Wrong approval, first we have to fix the bugs, after we can merge into dev, so please don't merge in the actual situation

**Review by Korab-Bejta — APPROVED** [review]

**Review by LYCChiang — APPROVED** [review]

**Review by GattoPianista — APPROVED** [review]

**Review by mdeliso97 — APPROVED** [review]
The corrections are good, still missin to fix the changing color/label of nodes is made also on selected edges

**Pull request merged**